src/RouterInjector.ts

```diff
@@ -1,87 +1,45 @@
-import { BaseInjector, Injector, InjectorProperties } from '@dojo/widget-core/Injector';
-import { DNode, RegistryLabel } from '@dojo/widget-core/interfaces';
-import { beforeRender } from '@dojo/widget-core/WidgetBase';
-import { w } from '@dojo/widget-core/d';
-import { WidgetRegistry } from '@dojo/widget-core/WidgetRegistry';
+import { Registry } from '@dojo/widget-core/Registry';
+import { Injector } from '@dojo/widget-core/Injector';
+import { RegistryLabel } from '@dojo/widget-core/interfaces';
 
 import HashHistory from './history/HashHistory';
 import { History } from './history/interfaces';
-import { MapParamsOptions, MatchType, OutletProperties, RouteConfig } from './interfaces';
 import { Router } from './Router';
+import { RouteConfig } from './interfaces';
 
 /**
  * Key for the router injector
  */
 export const routerKey = Symbol();
 
 /**
+ * Router Injector Options
+ *
+ */
+export interface RouterInjectorOptions {
+	history?: History;
+	key?: RegistryLabel;
+}
+
+/**
  * Creates a router instance for a specific History manager (default is `HashHistory`) and registers
  * the route configuration.
  *
  * @param config The route config to register for the router
  * @param registry An optional registry that defaults to the global registry
- * @param history The history manager the router needs to use, default is `HashHistory`
- * @param key The key for the router injector, defaults to exported `routerKey` symbol
+ * @param options The router injector options
  */
-export function registerRouterInjector(
-	config: RouteConfig[],
-	registry: WidgetRegistry,
-	history: History = new HashHistory(),
-	key: RegistryLabel = routerKey
-): Router<any> {
-	if (registry.has(key)) {
+export function registerRouterInjector(config: RouteConfig[], registry: Registry, options: RouterInjectorOptions = {}): Router<any> {
```

Comment:
the defaults aren't being tested, this line and the next... either they are always there or always not there

Reply:
Yup. Because for node we always have to use the MemoryHistory, could add a test that skips in node perhaps.

Reply:
minor nit really...

Reply:
@kitsonk do you want me to add the test to cover the defaults?

Reply:
Naw, it was a minor nit and the PR increases the overall coverage of the package.